server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -397,6 +398,64 @@ public TimeValue timeout() {
         }, onFailure));
     }
 
+    @Override
+    public void cloneShardSnapshot(SnapshotId source, SnapshotId target, RepositoryShardId shardId,
+                                   @Nullable String shardGeneration, ActionListener<String> listener) {
+        final IndexId index = shardId.index();
```

Comment:
I think we should check and fail that the repository is not read-only?

Reply:
We currently don't do that for `snapshotShard` but I guess it might be nice to have that safety measure here as well :) I'll add it to both places.

Reply:
Yes I think we should